### PR TITLE
DEV: allows for multiple menus/tooltips

### PR DESCRIPTION
**Pull request by jjaffeux**
menus and tooltips are now appended to their own portals. The service are the only responsible for managing the instances, prior to this commit, services could manage one instance, but the DMenu and DTooltip components could also take over which could cause unexpected states.

This change also allows nested menus/tooltips.

Other notable changes:
- few months ago core copied the `CloseOnClickOutside` modifier of float-kit without removing the float-kit one, this commit now only use the core one.
- the close function is now trully async
- the close function accepts an instance or an identifier as parameter

### CODE REVIEW

File: app/assets/javascripts/float-kit/addon/services/menu.js [hunk 1]
```diff
@@ -1,14 +1,13 @@
...
+import { schedule } from "@ember/runloop";
 import Service from "@ember/service";
+import { TrackedArray } from "@ember-compat/tracked-built-ins";
 import DMenuInstance from "float-kit/lib/d-menu-instance";
-import { updatePosition } from "float-kit/lib/update-position";
 
 export default class Menu extends Service {
-  @tracked activeMenu;
-  @tracked portalOutletElement;
+  @tracked registeredMenus = new TrackedArray();
```
Comment (CvX) [inline app/assets/javascripts/float-kit/addon/services/menu.js:10]: Does this need to be a `TrackedArray`? I can only see wholesale reassignments of `registeredMenus` prop.

**Review by CvX — APPROVED** [review]

File: app/assets/javascripts/float-kit/addon/components/d-tooltip.gjs [hunk 2]
```diff
@@ -15,33 +17,25 @@ export default class DTooltip extends Component {
   @service tooltip;
   @service internalTooltip;
 
-  @tracked tooltipInstance = null;
+  registerTrigger = modifier((element) => {
+    if (!this.tooltipInstance?.trigger) {
+      next(() => {
```
[the remark this answers is not in the document]
Comment (jjaffeux) [reply app/assets/javascripts/float-kit/addon/components/d-tooltip.gjs:22]: The world.

[the remark this answers is not in the document]
Comment (jjaffeux) [reply app/assets/javascripts/float-kit/addon/components/d-tooltip.gjs:22]: More seriously I had exceptions in tests I think

[the remark this answers is not in the document]
Comment (jjaffeux) [reply app/assets/javascripts/float-kit/addon/components/d-tooltip.gjs:22]: @CvX here is the exact error without the next:

global failure: Error: Assertion Failed: You attempted to update `_trigger` on `DTooltipInstance`, but it had already been used previously in the same computation.  Attempting to update a value after using it in a computation can cause logical errors, infinite revalidation bugs, and performance issues, and is not supported.

[the remark this answers is not in the document]
Comment (jjaffeux) [reply app/assets/javascripts/float-kit/addon/components/d-tooltip.gjs:22]: I will merge with the next for now, but open to a better solution if you know one

**Pull request merged**